presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +702,335 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+            if (token.isPresent()) {
+                tableSnapshotIdMap.put(handle, token.get().getSnapshotId());
+            }
+            else {
+                tableSnapshotIdMap.put(handle, (long) -1);
+            }
+        }
+        transactionHandle.setTableSnapshotIdMap(tableSnapshotIdMap);
+        return new IcebergWritableTableHandle(
+            table.getSchemaName(),
+            table.getTableName(),
+            SchemaParser.toJson(icebergTable.schema()),
+            PartitionSpecParser.toJson(icebergTable.spec()),
+            getColumns(icebergTable.schema(), typeManager),
+            getDataPath(icebergTable.location()),
+            getFileFormat(icebergTable));
+    }
+
+    @Override
+    public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(ConnectorSession session,
+            ConnectorInsertTableHandle insertHandle,
+            Collection<Slice> fragments,
+            Collection<ComputedStatistics> computedStatistics,
+            List<ConnectorTableHandle> sourceTableHandles)
+    {
+        IcebergWritableTableHandle table = (IcebergWritableTableHandle) insertHandle;
+        org.apache.iceberg.Table icebergTable = transaction.table();
+        List<CommitTaskData> commitTasks = new ArrayList<>();
+
+        if (!fragments.isEmpty()) {
+            commitTasks = fragments.stream()
+                .map(slice -> commitTaskCodec.fromJson(slice.getBytes()))
+                .collect(toImmutableList());
+
+            Type[] partitionColumnTypes = icebergTable.spec().fields().stream()
+                .map(field -> field.transform().getResultType(
+                    icebergTable.schema().findType(field.sourceId())))
+                .toArray(Type[]::new);
+
+            AppendFiles appendFiles = transaction.newFastAppend();
+            for (CommitTaskData task : commitTasks) {
+                DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
+                        .withInputFile(HadoopInputFile.fromLocation(task.getPath(), getConfiguration(session, table.getSchemaName())))
+                        .withFormat(table.getFileFormat())
+                        .withMetrics(task.getMetrics().metrics());
+
+                if (!icebergTable.spec().fields().isEmpty()) {
+                    String partitionDataJson = task.getPartitionDataJson()
+                            .orElseThrow(() -> new VerifyException("No partition data for partitioned table"));
+                    builder.withPartition(PartitionData.fromJson(partitionDataJson, partitionColumnTypes));
+                }
+
+                appendFiles.appendFile(builder.build());
+            }
+
+            // Update the map that tracks tables on which the materialized view depends and the corresponding snapshot ids of the table
+            String tableSnapshotIDString = Joiner.on(",").withKeyValueSeparator("=").join(transactionHandle.getTableSnapshotIdMap());
```

Comment:
Nit: we only capitalize the first letter of words, even acronyms, so `tableSnapshotIdString`

Reply:
We shouldn't depend on the `toString()` method of `IcebergTableHandle` as the serialization format, since it could change and it's not designed to be serializable. Using `SchemaTableName` directly and serializing to a JSON list seems better, since that handles names with special characters, etc.

Reply:
Now I am using `SchemaTableName.toString`. Not serializing to json since Iceberg stores this table property in JSON metadata.